### DISCUSSION TITLE
Model Refresh Methodology

### DISCUSSION THREAD

**Discussion by hdesgran** [google/meridian#248] [Q&A] [answered]
Hello,
I'm reaching out regarding the concept of model refresh discussed [here](https://developers.google.com/meridian/docs/advanced-modeling/refreshing-model). Specifically, it is mentionned : "When you append new data, you can set the priors so that they match the posterior for results before the appended data."

Is there any plan to implement this feature in future releases ?

Could you provide more detailed guidance on the major steps to follow in order to implement this methodology in a way that is compatible with Meridian ?

**Comment by mikewurm** [comment, marked as answer]
In short, no this is not something that would be implemented directly. However, the guidance in the documentation can be applied using the existing software.

Meridian only accepts parametric prior distributions. The guidance would be to choose a parametric ROI prior distribution that approximates the posterior distribution. For example, for each channel you can use an independent log-normal prior with the same mean and variance as the channel's previous posterior ([demonstrated in this code example](https://developers.google.com/meridian/docs/advanced-modeling/set-custom-priors-past-experiments)).

The guidance is not to precisely align the new prior with the old posterior, since this is not a standard Bayesian approach. As mentioned in the documentation, setting priors based on a previous MMM effectively uses some of the data twice (the data common to both model runs).

**Reply by AdimDrewnik** [reply]
Are there any plans to reconsider making a refresh function in Meridian?

**Reply by dattatreyam23** [reply]
Hi @AdimDrewnik  ,

Thank you for reaching out to the Google Meridian support team.

Meridian currently supports refreshing the model manually. When you append new data, you can set the priors so that they match the posterior for results before the appended data. See [Refresh the model](https://developers.google.com/meridian/docs/advanced-modeling/refreshing-model).

You may also refer to [Discussion #522](https://github.com/google/meridian/discussions/522#discussioncomment-12244949) that deals with a similar issue as yours, and has a detailed response on the same.

I understand that you are asking about an integrated refresh function for Meridian. We appreciate your feedback and shall take it into consideration for our upcoming releases. Our team is constantly working on improving Meridian's functionality and you may check the unreleased features on the [CHANGELOG](https://github.com/google/meridian/blob/60e943ed3b2cabea5e722e5f696c9fb9ccc4025c/CHANGELOG.md) page in GitHub.

Feel free to reach out if you have any questions or suggestions regarding Meridian.

Thank you,

Google Meridian Support Team
